sources/packages/backend/libs/test-utils/src/factories/application.ts

```diff
@@ -210,6 +210,9 @@ export async function saveFakeApplicationDisbursements(
   savedApplication.currentAssessment.studentAssessmentStatus =
     options?.currentAssessmentInitialValues?.studentAssessmentStatus ??
     savedApplication.currentAssessment.studentAssessmentStatus;
+  savedApplication.currentAssessment.triggerType =
+    options?.currentAssessmentInitialValues?.triggerType ??
+    savedApplication.currentAssessment.triggerType;
```

Comment:
There is a concern of cognitive complexity raised by sonar cloud analysis. Please look into it.

Reply:
To ensure that Sonar Cloud has no issues, I have removed lines 213 to 215 and added the code `application.currentAssessment.triggerType = AssessmentTriggerType.RelatedApplicationChanged` in its corresponding test cases.